packages/org.openapis.v3/SecurityScheme.pkl

```diff
@@ -30,10 +31,10 @@ type: SecuritySchemeType
 description: String?
 
 /// The name of the header, query or cookie parameter to be used.
-name: String
+name: String?((type == "apiKey").implies(this != null))
```

Comment:
I think this condition may be reversed. As-is, it doesn't forbid these properties from being non-null when the type is not `apiKey`.

```suggestion
name: String?((this != null).implies(type == "apiKey"))
```